code/modules/atmospherics/machinery/components/binary_devices/circulator.dm

```diff
@@ -90,8 +90,48 @@
 	if(generator)
 		disconnectFromGenerator()
 	to_chat(user, "<span class='notice'>You [anchored?"secure":"unsecure"] [src].</span>")
+
+	var/obj/machinery/atmospherics/node1 = nodes[1]
+	if(node1)
+		node1.disconnect(src)
+		nodes[1] = null
+		nullifyPipenet(parents[1])
+	var/obj/machinery/atmospherics/node2 = nodes[2]
+	if(node2)
+		node2.disconnect(src)
+		nodes[2] = null
+		nullifyPipenet(parents[2])
+
+	if(anchored)
+		SetInitDirections()
+		atmosinit()
+		node1 = nodes[1]
+		if(node1)
+			node1.atmosinit()
+			node1.addMember(src)
+		node2 = nodes[2]
+		if(node2)
+			node2.atmosinit()
+			node2.addMember(src)
+		build_network()
+
 	return TRUE
 
+/obj/machinery/atmospherics/components/binary/circulator/SetInitDirections()
+	switch(dir)
+		if(NORTH, SOUTH)
+			initialize_directions = EAST|WEST
+		if(EAST, WEST)
+			initialize_directions = NORTH|SOUTH
+
+/obj/machinery/atmospherics/components/binary/circulator/getNodeConnects()
+	return list(turn(dir, 90), turn(dir, 270))

```

Comment:
this'll result in the the TEG looking in the wrong direction for nodes, since the first entry is the input and the second is the output. It would look something like this:

![image](https://user-images.githubusercontent.com/20558591/41786717-92735490-7646-11e8-991a-f0244e880a6d.png)

Reply:
Now you can flip circulator and make TEG of your dream.